### PR TITLE
Keyword arguments for Matplotlib.pyplot.quiver

### LINKED ISSUE #11526: Matplotlib.pyplot.quiver does not accept X, Y, U, V and C as keyword arguments
### Bug report

**Bug summary**

Bug: matplotlib.pyplot.quiver cannot accept X, Y, U, V and C as keyword arguments, such operation leads to an `IndexError: pop from empty list` (if all of them are keyword arguments) or an `AttributeError: Unknown property *` (if part of them are keyword arguments). This error arises because line 444 in `quiver.py`, inside the constructor of Quiver class, only uses `*args` but excludes `**kw` for parsing of arguments X, Y, U, V and C, and thus nothing can be parsed from `*args` in the function `_parse_args` on line 385 if these arguments are given by keywords.

**Code for reproduction**

```python
import matplotlib.pyplot as plt

x, y, u, v = 0, 0, .01, .02

plt.quiver(X=x, Y=y, U=u, V=v, units='xy', scale=1)
plt.show()
```

**Actual outcome**

```
Traceback (most recent call last):
  File "untitled.py", line 5, in <module>
    plt.quiver(X=x, Y=y, U=u, V=v, units='xy', scale=1)
  File "/usr/local/lib/python3.6/site-packages/matplotlib/pyplot.py", line 3425, in quiver
    ret = ax.quiver(*args, **kw)
  File "/usr/local/lib/python3.6/site-packages/matplotlib/__init__.py", line 1855, in inner
    return func(ax, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/matplotlib/axes/_axes.py", line 4872, in quiver
    q = mquiver.Quiver(self, *args, **kw)
  File "/usr/local/lib/python3.6/site-packages/matplotlib/quiver.py", line 444, in __init__
    X, Y, U, V, C = _parse_args(*args)
  File "/usr/local/lib/python3.6/site-packages/matplotlib/quiver.py", line 393, in _parse_args
    V = np.atleast_1d(args.pop(-1))
IndexError: pop from empty list
```
And the error disappears if we write the code as `plt.quiver(x, y, u, v, units='xy', scale=1)`.

**Matplotlib version**
  * Operating system: Mac OS High Sierra
  * Matplotlib version: 2.2.2 (installed from pip3)
  * Matplotlib backend: MacOSX
  * Python version: 3.6.5

### PR DESCRIPTION
**Pull request by jaideep-seth**
## PR Summary

Closes #11526
## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by jaideep-seth** [comment]
#11526 
 this is my first attempt at Matplotlib, is the PR alright?

-Thank you!

**Comment by jklymak** [comment]
Thanks so much for the PR.  This really needs to pass the tests, and probably needs some new tests.  If at all possible, setup the testing environment for your self, and run some of the tests that have failed (drill down in `continuous-integration/travis-ci/pr`, is usually the most straightforward).  Its usually easiest to run one or two tests at a time i.e. 

```
pytest -v lib/matplotlib/tests/test_tightlayout.py::test_badsubplotgrid
```

just runs that one test.  

- https://matplotlib.org/devel/testing.html#testing
- https://matplotlib.org/devel/contributing.html

Let us know if you have any questions.

**Comment by jklymak** [comment]
Also, please fill out the PR description.

File: lib/matplotlib/quiver.py [hunk 1]
```diff
@@ -381,27 +380,60 @@ def contains(self, mouseevent):
...
+            nr, nc = U.shape
+        if len(args) == 2:  # remaining after removing U,V,C CASE-1
+            X, Y = [np.array(a).ravel() for a in args]
+            if len(X) == nc and len(Y) == nr:
+                X, Y = [a.ravel() for a in np.meshgrid(X, Y)]
+        else:#CASE 2
+            indexgrid = np.meshgrid(np.arange(nc), np.arange(nr))
+            X, Y = [np.ravel(a) for a in indexgrid]
+    
+    elif len(args) == 0:#only keyword arguments
```
Comment (jklymak) [inline lib/matplotlib/quiver.py]: Are there really only these two possibilities?  What if the user provides both kwargs and positional args?

**Comment by jaideep-seth** [comment]
Hey, thank you for the guidance!
-  i deleted my repository and cloned the matplotlib fork again, after following all the instructions on: https://matplotlib.org/faq/installing_faq.html#install-from-git   i ran 'py.test' to see if all the tests were passing before i made any changes and  'ImportError: cannot import name 'ft2font' is the error i get'. Please help me!!

- also matplotlib/doc-requirements.txt doesnt have all requirements like 'pytest'.

**Comment by anntzer** [comment]
Closing per decision in #11526.

**Pull request closed without merging**